crates/verifier/src/error/span.rs

```diff
@@ -0,0 +1,81 @@
+use std::fmt;
+
+use sonatina_ir::Type;
+
+use super::Stacktrace;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum Span {
+    Function,
+    Block,
+    Insn,
+    Callee,
+    Value,
+    Type,
+    GlobalVariable,
+    CompoundType,
+    _StackLimit,
+}
```

Comment:
The term "Span" is generally used to refer to a specific range in the source code (start and end positions), so this can be a confusing expression. A name like `IrNodeKind` would fit well.

Also, this struct doesn't seem to be used effectively. It seems `ErrorKind` and `StackTrace` already contains all the necessary information, and the generated error messages don't appear to make effective use of this struct. It might be better to remove this part for now until it turns out that we need a more fine-grained location of an error message.

Reply:
I thought about adding an optional parent ir node here. like parent instruction, or none if it's the first instruction in block. yes, will leave it out of this pr.

Reply:
hm, the span will also print the opaque ref for insn, cmpd type, etc. should I combine this into trace info string?

Reply:
> I thought about adding an optional parent ir node here. like parent instruction, or none if it's the first instruction in block

It'd be not so efficient(I want to avoid a linked list representation here). We probably need to think about source mapping(i.e., a mapping between a sonatina IR and an original source mapping), as well. It's still unclear how we can encode those information in sonatina. 
So yeah, let's put them on the back burner for now.

Reply:
> hm, the span will also print the opaque ref for insn, cmpd type, etc. should I combine this into trace info string?
 
 I don't think we need opaque references as displayed information (it's just an integer), so it'd be fine not to display them.